iroh/src/node.rs

```diff
@@ -308,9 +308,22 @@ impl<D: iroh_blobs::store::Store> NodeInner<D> {
                 },
                 // handle task terminations and quit on panics.
                 res = join_set.join_next(), if !join_set.is_empty() => {
-                    if let Some(Err(err)) = res {
-                        error!("Task failed: {err:?}");
-                        break;
+                    match res {
+                        Some(Err(outer)) => {
+                            if outer.is_panic() {
+                                error!("Task panicked: {outer:?}");
+                                break;
+                            } else if outer.is_cancelled() {
+                                debug!("Task cancelled: {outer:?}");
+                            } else {
+                                error!("Task failed: {outer:?}");
+                                break;
+                            }
+                        }
```

Comment:
Do we need to print cancellations?
And IIUC, the third case is impossible. It's either a cancel or a panic: https://docs.rs/tokio/latest/src/tokio/runtime/task/error.rs.html#15-18

```suggestion
                        Some(Err(outer)) if outer.is_panic() => {
                            error!("Task panicked: {outer:?}");
                        }
```

Reply:
> Do we need to print cancellations? And IIUC, the third case is impossible. It's either a cancel or a panic: https://docs.rs/tokio/latest/src/tokio/runtime/task/error.rs.html#15-18

Yeah, I thought so as well, but then why don't they expose an enum? The way the JoinError API is written it seems that they want to keep the possibility open to have a third case.

Reply:
> Do we need to print cancellations?

Not sure if they would ever happen at all. But if it happens, I would like to know.